Engine/Src/Ancona/Menu/Systems/ButtonSystem.cpp

```diff
@@ -0,0 +1,230 @@
+#include <sstream>
+
+#include "ButtonSystem.hpp"
+
+using namespace ildmenu;
+
+ButtonStateEnum GetNextButtonState(ButtonStateEnum currentState, PointerStateEnum pointerState, bool mouseIsOver);
+PointerStateEnum GetNextPointerState(bool isDown, PointerStateEnum previousState);
+std::string GetButtonStateTitle(ButtonStateEnum buttonState);
+
+
+/* Component */
+ButtonComponent::ButtonComponent()
+{
+}
+
+void ButtonComponent::Update(float delta, const ild::Box2 & position, PointerStateEnum pointerState)
+{
+
+    // Implement component logic here
+    auto mouseIsOver = position.Intersects(_collisionComponent->box());
+    auto initalState = _buttonState;
+    _buttonState = GetNextButtonState(_buttonState, pointerState, mouseIsOver);
+
+
+    if (initalState != _buttonState)
+    {
+        //TODO Gaurd against null with asserts
```

Comment:
we should either do what's in the TODO or write an issue and refer to it. Also you spelled "guard" wrong